llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -304,10 +304,20 @@ class YkControlPoint : public ModulePass {
             ->getType();
 
     // Create the new control point, which is of the form:
-    // bool new_control_point(YkMT*, YkLocation*, CtrlPointVars*)
-    FunctionType *FType = FunctionType::get(
-        Type::getInt1Ty(Context),
-        {YkMTTy, YkLocTy, CtrlPointVarsTy->getPointerTo()}, false);
+    //   bool new_control_point(YkMT*, YkLocation*, CtrlPointVars*, ReturnValue*)
+    // If the control points caller return type is void, create an Int1 pointer
```

Comment:
s/caller return type/return type/ ? [I'm not sure what "caller return type" is but maybe it means something special?]

Reply:
also s/points/point's/

Reply:
It's the return type of the function in which the control point is, e.g.:
```
int main() {
   yk_control_point()
}
```
Here, the caller of the control point is `main` and it's type is `int`. Happy to take suggestions for a better name.

Reply:
Oh I see! I think you'll struggle to find an unambiguous term for this, so it's worth spelling it out in full: "(i.e. for a function f that calls yk_control_point, the return type of f)" or similar.

Reply:
Try e510ad84

Reply:
"Its" is a dangerous word and it's ambiguous in that commit "its return type is void". Which "it"? I was careful to avoid "if" in my suggestion for that reason! If we can get rid of "it" I think we'll be good to go.

Reply:
Fixed in a55604